### DISCUSSION TITLE
Is documentation about supported tokens for exit format up to date?

### DISCUSSION THREAD

**Discussion by ssa3512** [statechannels/go-nitro#1274] [General]
Currently the documentation states that "At the current time, Nitro supports native (e.g. ETH) and ERC20 tokens."
https://github.com/statechannels/go-nitro/blob/fe6ffa430123792b86c675b160d7c17f5a8fb7b1/docs/protocol-tutorial/0030-outcomes.md?plain=1#L11-L13

but it appears that exit format support for ERC721 and ERC1155 support were added here:
https://github.com/statechannels/exit-format/pull/69

Can you provide clarification on if the documentation is accurate or if if should be updated to reflect support for 721/1155 tokens?

**Comment by geoknee** [comment]
Hey @ssa3512 thanks for bringing this up. 

You are correct that support for more tokens was added to `exit-format`. However, that increased functionality was not yet bubbled up to `go-nitro`. You can see from the deposit function on our adjudicator: 

https://github.com/statechannels/go-nitro/blob/267bf7dde944f26a786d97c18aa5e5a59d543ba6/nitro-protocol/contracts/MultiAssetHolder.sol#L29-L63

This means the docs are accurate (of course we could make things clearer by heading off this question in the docs). 

One thing that makes full integration of the exit-format a little awkward is that in our on chain code, we have a special escape hatch for what we call  internal / external destinations:

https://github.com/statechannels/go-nitro/blob/267bf7dde944f26a786d97c18aa5e5a59d543ba6/nitro-protocol/contracts/MultiAssetHolder.sol#L406-L423

So we aren't currently relying on the `executeExit()` function from `exit-format`. I think it should be possible to fix all of this -- it's not currently a priority for us, though.
